docs/site/Interceptors.md

```diff
@@ -0,0 +1,529 @@
+  // This is needed if a custom decorator is created for `@intercept`
+  @intercept(log)
+  @intercept(logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // Apply multiple interceptors. The order of `log` will be preserved as it
+  // explicitly listed at method level
+  @intercept(convertName, log)
+  async greet(name: string) {
+    return `Hello, ${name}`;
+  }
+}
+```
+
+### Global interceptors
+
+Global interceptors are discovered from the `InvocationContext`. They are
+registered as bindings with `interceptor` tag. For example,
+
+```ts
+import {asInterceptor} from '@loopback/context';
+
+app
+  .bind('interceptors.MetricsInterceptor')
+  .toProvider(MetricsInterceptorProvider)
+  .apply(asInterceptor);
+```
+
+### Order of invocation for interceptors
+
+Multiple `@intercept` decorators can be applied to a class or a method. The
+order of invocation is determined by how `@intercept` is specified. The list of
+interceptors is created from top to bottom and from left to right. Duplicate
+entries are removed from their first occurrences.
+
+Let's examine the list of interceptors invoked for each method on
+`MyControllerWithClassLevelInterceptors`, which has a class level `log`
+decorator:
+
+```ts
+// Apply `log` to all methods on the class
+@intercept(log)
+class MyControllerWithClassLevelInterceptors {
+  // ...
+}
+```
+
+1. greetStatic: [`log`]
+
+   ```ts
+   // No explicit `@intercept` at method level. The class level `log` will
+   // be applied
+   static async greetStatic(name: string) {
+     return `Hello, ${name}`;
+   }
+   ```
+
+2. greetStaticWithDI: [`log`]
+
+   ```ts
+   // The method level `log` overrides the class level one
+   @intercept(log)
+   static async greetStaticWithDI(@inject('name') name: string) {
+     return `Hello, ${name}`;
+   }
+   ```
+
+3. greetSync: [`log`, `logSync`]
+
+   ```ts
+   // We can apply `@intercept` multiple times on the same method
+   // This is needed if a custom decorator is created for `@intercept`
+   @intercept(log) // The method level `log` overrides the class level one
+   @intercept(logSync)
+   greetSync(name: string) {
+     return `Hello, ${name}`;
+   }
+   ```
+
+4. greet: [`convertName`, `log`]
+
+   ```ts
+   // Apply multiple interceptors. The order of `log` will be preserved as it
+   // explicitly listed at method level
+   @intercept(convertName, log)
+   async greet(name: string) {
+     return `Hello, ${name}`;
+   }
+   ```
+
+## Create your own interceptors
+
+Interceptors can be made available by LoopBack itself, extension modules, or
+applications. They can be a function that implements `Interceptor` signature or
+a binding that is resolved to an `Interceptor` function.
+
+### Interceptor functions
+
+The interceptor function has the following signature:
+
+```ts
+export interface Interceptor {
+  <T = unknown>(
+    context: InvocationContext,
+    next: () => ValueOrPromise<T>,
+  ): ValueOrPromise<T>;
+}
```

Comment:
We will need to update the signature of this function based on the comments I posted in the TypeScript source code.